Sources/BrowserServicesKit/InternalUserDecider/InternalUserDecider.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  InternalUserDecider.swift
+//  DuckDuckGo
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public protocol InternalUserDecider {
+    
+    var isInternalUser: Bool { get }
+    
+    @discardableResult
+    func markUserAsInternalIfNeeded(forUrl url: URL?, response: HTTPURLResponse?) -> Bool
+}
+
+public protocol InternalUserStore {
+    var didVerifyInternalUser: Bool { get set }
+}
+
+public class DefaultInternalUserDecider: InternalUserDecider {
+    private let userDefaults = UserDefaults()
```

Comment:
This may be controversial as I can only see one other place in BSK that manages UserDefaults access itself and not through an injected store. This change would also meant that everyone would have to re-authenticate which would be mildly inconvenient for internal users only.

Reply:
Actually need to inject a store as it's encrypted on macOS. Shouldn't be too hard

Reply:
At any rate, I would definitely rather avoid people having to reauthenticate if possible, we have a good adoption rate at the moment, and I'm not sure how easy that will be to replicate.